agent/recorder.go

```diff
@@ -343,12 +343,17 @@ func (r *SpanRecorder) getPayload(rawSpans []tracer.RawSpan) map[string]interfac
 			})
 		}
 	}
-	return map[string]interface{}{
-		"metadata":   r.metadata,
+	payload := map[string]interface{}{
 		"spans":      spans,
 		"events":     events,
 		tags.AgentID: r.agentId,
 	}
+	// We send the metadata only if not spans are sent successfully
+	if atomic.LoadInt64(&r.stats.sendSpansOk) == 0 {
```

Comment:
So if sendSpansOK > 0, the `metadata` will not be sent, right? 
Is this already supported in the backend?

Reply:
Not released yet in the backend, I made this `PR` because @AdrianLC need it to test the feature in the backend. We will merge it when the backend gets ready.